src/RestApi.php

```diff
@@ -55,6 +56,37 @@ public static function get_routes_from_namespace( $namespace ) {
 		return isset( $response_data['routes'] ) ? $response_data['routes'] : null;
 	}
 
+	/**
+	 * Check if is request to the Store API.
+	 *
+	 * @return bool
+	 */
+	protected static function is_request_to_store_api() {
+		if ( empty( $_SERVER['REQUEST_URI'] ) ) {
+			return false;
+		}
+
+		$rest_prefix = trailingslashit( rest_get_url_prefix() );
+		$request_uri = esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+		return false !== strpos( $request_uri, $rest_prefix . 'wc/store' );
+	}
+
+	/**
+	 * The Store API does not require authentication.
+	 *
+	 * @param \WP_Error|mixed $result Error from another authentication handler, null if we should handle it, or another value if not.
+	 * @return WP_Error|mixed|bool
```

Comment:
I think this is supposed to be `WP_Error|null|bool` (from the filter definition in WP core).